docs/RFCS/20210527_changefeed_http_sink.md

```diff
@@ -0,0 +1,202 @@
+  base64-encoded auth or an OAuth token. As this contains credentials, it
+  should be redacted when appearing in the `description` column of `SHOW JOBS`/
+  `SHOW CHANGEFEED JOBS` etc.
+* `http_client_timeout` - specifies a timeout (in seconds) for the client to 
+  wait for each request to complete (corresponds to Go's 
+  [`http.client.Timeout`](https://golang.org/src/net/http/client.go#L105)) 
+  If no option is provided, the default timeout is 3 seconds.
+
+Altogether, the command to create a changefeed to an HTTP sink would look 
+something like this:
+
+```
+$ root@:26257/defaultdb> CREATE CHANGEFEED FOR foo INTO 
+                      -> INTO 'https://fake-endpoint:8080?ca_cert=Zm9v...'
+                      -> WITH http_auth_header = 'Bearer <token>',
+                      -> WITH http_client_timeout = 5;
+```
+
+## Testing Scenarios
+
+1. Normally operating HTTP sink
+    * Reception by mock server
+    * Sink flushed properly
+2. Normally operating HTTP sink with TLS (ca cert)
+3. Normally operating HTTP sink with options
+   * `http_auth_header` - basic auth & OAuth works for correct credentials,
+     fails for wrong credentials
+   * `http_client_timeout` - setup mock server to sleep longer than provided 
+     timeout, observe error
+4. Server provides unknown cert (no `ca_cert` specified by user)
+    * Observe TLS error
+    * Observe success with `insecure_tls_skip_verify` enabled
+5. Improper options (wrong type, options missing, incompatible options),
+   continuation of `TestChangefeedErrors()` in `changefeed_test.go`
+5. Jobs table properly redacts credentials
+    * Create changefeed with `http_auth_header` option set
+    * Check that running SHOW JOBS/SHOW CHANGEFEED JOBS has the `description` 
+      column redacted
+
+## Rationale and Alternatives
+
+A webhook-type HTTP sink was decided on compared to other approaches due to is
+simplicity and ease of implementation. Connections do not have to be maintained
+or closed and can remain stateless.
+
+In terms of client authentication, customers expressed interest in basic
+authentication via HTTP headers, this method is more common and easier to setup,
+however client certs can be added as well (`client_cert` param passed to Kafka).
+Can easily add this as an option if needed.
+
+# Unresolved questions/Future additions
+
+* Creating another HTTP sink may cause ambiguity, especially for users of the
+  old HTTP cloud storage sink. For now, the current naming convention should 
+  work (`http(s)` -> new http sink and `experimental-http(s)` -> old cloud 
+  storage sink), however there are plans to 
+  [drop](https://github.com/cockroachdb/cockroach/issues/53716) the 
+  `experimental-` prefix from the cloud storage sink so perhaps an option 
+  should be added to use the old sink when this is done. Goal is to be backward
+  compatible but also encourage new users to use the new sink as it's more 
+  likely to be in line with their expectations.
```

Comment:
Something that we've brought up before is that currently we mix the "transport" with a notion of the "framing" or "application protocol" by which we send updates.  I think this more or less made sense for the kafka sink, but definitely starts making less sense when we talk about more generic transports like HTTP.